src/nvml/hook.c

```
@@ -353,16 +353,20 @@ nvmlReturn_t _nvmlDeviceGetMemoryInfo(nvmlDevice_t device,void* memory,int versi
             return NVML_SUCCESS;
         }
     } else {
+        if (usage > limit) {
+            LOG_WARN("usage(%ld) exceeds limit(%ld), clamping", usage, limit);
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)